apps/omg_watcher_info/lib/omg_watcher_info/db/pending_block.ex

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.DB.PendingBlock do
+  @moduledoc """
+  Ecto schema for pending block data
+  These are valid block data received by the internal bus that should be stored in the database.
+  This intermediate table is needed as the messages received by the bus are not persisted
+  and if for some reason the writing to the database fails, we would lose these data.
+  """
+  use Ecto.Schema
+  use OMG.Utils.LoggerExt
+
+  import Ecto.Changeset
+  import Ecto.Query, only: [from: 2]
+
+  alias OMG.WatcherInfo.DB.Repo
+
+  @type t() :: %{
+          blknum: pos_integer(),
+          data: binary(),
+          status: String.t()
+        }
+
+  @status_pending "pending"
```

Comment:
I don't think you need these two states. If it's in the queue, it's pending. Once it's processed, it's removed from the queue.

Reply:
Yeah, I was wondering if we should keep them, was waiting for your opinion. I'll delete them and remove the status.